crates/cli/src/helper/lock.rs

```diff
@@ -236,6 +245,167 @@ pub fn path_to_pkg_name(path_str: &str) -> Option<&str> {
     }
 }
 
+pub async fn validate_deps() -> Result<()> {
+    // Read package.json for overrides
+    let pkg_file = load_package_json()?;
+    // Initialize overrides
+    let overrides = Overrides::new(pkg_file.clone()).parse(pkg_file.clone());
+
+    let lock_file = load_package_lock_json()?;
+
+    // check package-lock.json packages and package.json dependencies are the same
+    let pkg_in_pkg_lock = lock_file
+        .get("packages")
+        .and_then(|p| p.as_object())
+        .and_then(|p| p.get(""));
+    if let Some(root_pkg) = pkg_in_pkg_lock {
+        for (dep_field, _is_optional) in get_dep_types() {
+            if root_pkg.get(dep_field) != pkg_file.get(dep_field) {
+                return Err(anyhow::anyhow!("{} changed", dep_field));
+            }
+        }
+    }
+
+    if let Some(packages) = lock_file.get("packages").and_then(|p| p.as_object()) {
+        for (pkg_path, pkg_info) in packages {
+            for (dep_field, is_optional) in get_dep_types() {
+                if let Some(dependencies) = pkg_info.get(dep_field).and_then(|d| d.as_object()) {
+                    for (dep_name, req_version) in dependencies {
+                        let req_version_str = req_version.as_str().unwrap_or_default();
+
+                        // Collect parent chain information
+                        let mut parent_chain = Vec::new();
+                        let mut current_path = if pkg_path.is_empty() {
+                            String::new()
+                        } else {
+                            pkg_path.to_string()
+                        };
```

Comment:
How about the following code:

```suggestion
 let mut current_path = String::from(pkg_path);
```